src/Microsoft.AspNet.Mvc.DataAnnotations/DataTypeAttributeAdapter.cs

```diff
@@ -39,5 +39,10 @@ public DataTypeAttributeAdapter(DataTypeAttribute attribute, string ruleName, IS
             var errorMessage = GetErrorMessage(context.ModelMetadata);
             return new[] { new ModelClientValidationRule(RuleName, errorMessage) };
         }
+
+        public string GetErrorMessage(ModelMetadata metadata)
```

Comment:
Oops ... all the other `DataAnnotationsClientModelValidator<TAttribute>` subclasses.

More generally it's weird having the server-side stack calling into a `DataAnnotationsClientModelValidator<TAttribute>` subclass. Calling into an `IClientModelValidator` implementation is also a bit odd but not as bad. Suggest at least renaming `DataAnnotationsClientModelValidator<TAttribute>` e.g. to `ValidationAttributeAdapter<TAttribute>`.